### PR TITLE
task/DES-2562: Add `Scratch (Frontera)` to listings

### PR DESCRIPTION
**Pull request by jarosenb**
## Overview: ##
Add `Scratch (Frontera)` as a nav option in the Data Depot and Workspace.

## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [DES-2562](https://jira.tacc.utexas.edu/browse/DES-2562)

## Testing Steps: ##
1. Navigate to Scratch (Frontera) in the Data Depot
2. in the Workspace, select Scratch (Frontera)) from the file browser.

## UI Photos:
image
image

### CODE REVIEW

**Comment by jarosenb** [comment]
No longer needed, we are updated `desigsnafe.storage.default` directly.

File: designsafe/static/scripts/data-depot/components/data-depot-nav/data-depot-nav.component.js [hunk 1]
```diff
@@ -45,6 +45,12 @@ class DataDepotNavCtrl {
           state: 'myData',
           description: 'Private directory for your data'
         },
+        {
+          name: 'My Working Files',
+          collapsible: false,
+          state: 'myDataWork',
+          description: 'Working directory for use with JupyterHub'
```
Comment (rstijerina) [inline designsafe/static/scripts/data-depot/components/data-depot-nav/data-depot-nav.component.js:52]:
```suggestion
          name: 'Scratch (Frontera)',
          collapsible: false,
          state: 'myDataScratch',
          description: 'Scratch directory on Frontera for use with JupyterHub'
```

File: designsafe/static/scripts/data-depot/components/data-depot-toolbar/data-depot-toolbar.component.js [hunk 2]
```diff
@@ -16,6 +16,7 @@ class DataDepotToolbarCtrl {
     placeholder() {
         var stateNames = {
             'myData': 'My Data',
+            'myDataWork': 'My Working Files',
```
Comment (rstijerina) [inline designsafe/static/scripts/data-depot/components/data-depot-toolbar/data-depot-toolbar.component.js:19]:
```suggestion
            'myDataScratch': 'Scratch (Frontera)',
```

File: designsafe/static/scripts/data-depot/index.js [hunk 3]
```diff
@@ -83,6 +83,39 @@ function config(
                 },
             },
         })
+        .state('myDataWork', {
+            url: '/agave/{systemId}/{filePath:any}/?query_string&offset&limit',
+            component: 'dataDepotBrowser',
+            params: {
+                systemId: 'designsafe.storage.working',
```
Comment (rstijerina) [inline designsafe/static/scripts/data-depot/index.js:90]:
```suggestion
        .state('myDataScratch', {
            url: '/agave/{systemId}/{filePath:any}/?query_string&offset&limit',
            component: 'dataDepotBrowser',
            params: {
                systemId: 'designsafe.storage.frontera.scratch',
```

File: designsafe/static/scripts/ng-designsafe/services/file-operation-service.js [hunk 4]
```diff
@@ -49,7 +49,7 @@ export class FileOperationService {
 
     getTests(files) {
         const externalDataStates = ['boxData', 'dropboxData', 'googledriveData'];
-        const agaveDataStates = ['myData', 'projects.view', 'projects.curation'];
+        const agaveDataStates = ['myData', 'myDataWork', 'projects.view', 'projects.curation'];
```
Comment (rstijerina) [inline designsafe/static/scripts/ng-designsafe/services/file-operation-service.js:52]:
```suggestion
        const agaveDataStates = ['myData', 'myDataScratch', 'projects.view', 'projects.curation'];
```

File: designsafe/static/scripts/workspace/components/workspace-data-browser/workspace-data-browser.component.js [hunk 5]
```diff
@@ -17,6 +17,7 @@ class WorkspaceDataBrowserCtrl {
 
         this.options = [
             { name: 'myData', label: 'My Data' },
+            { name: 'myDataWork', label: 'My Working Files' },
```
Comment (rstijerina) [inline designsafe/static/scripts/workspace/components/workspace-data-browser/workspace-data-browser.component.js:20]:
```suggestion
            { name: 'myDataScratch', label: 'Scratch (Frontera)' },
```

File: designsafe/static/scripts/workspace/components/workspace-data-browser/workspace-data-browser.component.js [hunk 6]
```diff
@@ -58,6 +59,17 @@ class WorkspaceDataBrowserCtrl {
                 });
                 this.breadcrumbParams = this.FileListingService.fileMgrMappings.agave.breadcrumbParams;
                 break;
+            case 'myDataWork':
+                this.listingType = 'files';
+                this.FileListingService.browse({
+                    section: 'main',
+                    api: 'agave',
+                    scheme: 'private',
+                    system: 'designsafe.storage.working',
```
Comment (rstijerina) [inline designsafe/static/scripts/workspace/components/workspace-data-browser/workspace-data-browser.component.js:68]:
```suggestion
            case 'myDataScratch':
                this.listingType = 'files';
                this.FileListingService.browse({
                    section: 'main',
                    api: 'agave',
                    scheme: 'private',
                    system: 'designsafe.storage.frontera.scratch',
```

**Review by rstijerina — CHANGES_REQUESTED** [review]
Updated for `designsafe.storage.frontera.scratch`

**Pull request merged**